xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFFmpegArchive.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (C) 2018 Arthur Liberman
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DVDInputStreamFFmpegArchive.h"
+#include "ServiceBroker.h"
+#include "pvr/PVRManager.h"
+#include "pvr/PVRStreamProperties.h"
+#include "pvr/addons/PVRClient.h"
+#include "utils/log.h"
+
+CDVDInputStreamFFmpegArchive::CDVDInputStreamFFmpegArchive(const CFileItem& fileitem)
+  : CDVDInputStreamFFmpeg(fileitem),
+    m_client(CServiceBroker::GetPVRManager().GetClient(fileitem))
```

Comment:
this is a super ugly hack and continues the old pvr inputstream crap. crap because it is a singleton, all streaming relevant functions are routed through the pvr addon instance. because of this thumbnails do not work for pvr recordings and picture-in picture is not possible. 
pvr addons should leave all streaming related functions to inputstream addons. those support multiple instances of the same type.

Reply:
> this is a super ugly hack and continues the old pvr inputstream crap

 But this "crap" is best we currently have. Everthing else is far far in the future, especially with noone in the team anymore with the skills required for such a big refactoring/rewrite.

Reply:
The inputstream api is there and pvr addons can delegate handling to those addons. All there and ready to use. Not need to continue with this old crap.

Reply:
Do we have any sample code for this? This would help a lot. I for myself have no clue how to gain access to other addons from inside a PVR addon and how to delegate something to them.

Reply:
checkout pvr.simple, it delegates stream handling to inputstream.adaptive https://github.com/kodi-pvr/pvr.iptvsimple/blob/1fdabd8863f24712db93d5e01a2953ab6a6bfb91/src/client.cpp#L278

you could write inputstream.tvh and pass all info needed by this addon to connect to the tvh backend with the fileitem.